staking-bank/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<u64> {
+    let key: PublicKeyEd25519 = ctx.parameter_cursor().get()?;
+
+    if is_validator(key) {
+        Ok(ONE)
+    } else {
+        Ok(0u64)
+    }
+}
+
+/// View function that returns a true, if all of the provided public keys are validators, otherwise a false.
+#[cfg(any(feature = "production", feature = "development", feature = "sandbox"))]
+#[receive(
+    contract = "staking_bank",
+    name = "verifyValidators",
+    parameter = "Vec<PublicKeyEd25519>",
+    return_value = "bool"
+)]
+fn verify_validators<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<bool> {
+    let keys: Vec<PublicKeyEd25519> = ctx.parameter_cursor().get()?;
+
+    for validator in keys {
+        if !is_validator(validator) {
+            return Ok(false);
+        }
+    }
+
+    Ok(true)
+}
+
+/// View function that returns the number of validtors.
+#[cfg(any(feature = "production", feature = "development", feature = "sandbox"))]
+#[receive(
+    contract = "staking_bank",
+    name = "getNumberOfValidators",
+    return_value = "u8"
+)]
+fn get_number_of_validators<S: HasStateApi>(
+    _ctx: &impl HasReceiveContext,
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<u8> {
+    Ok(NUMBER_OF_VALIDATORS)
+}
+
+/// View function that returns the balances of validators.
+#[cfg(any(feature = "production", feature = "development", feature = "sandbox"))]
+#[receive(
+    contract = "staking_bank",
+    name = "getBalances",
+    return_value = "Vec<u64>"
+)]
+fn get_balances<S: HasStateApi>(
+    _ctx: &impl HasReceiveContext,
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<Vec<u64>> {
+    let mut balances = Vec::with_capacity(NUMBER_OF_VALIDATORS as usize);
+    for _i in 0..NUMBER_OF_VALIDATORS {
+        balances.push(ONE)
+    }
+
+    Ok(balances)
+}
+
+/// View function that returns the public key of a validator from an index.
+#[cfg(any(feature = "production", feature = "development", feature = "sandbox"))]
+#[receive(
+    contract = "staking_bank",
+    name = "publicKey",
+    parameter = "u8",
+    return_value = "PublicKeyEd25519"
+)]
+fn public_key<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<PublicKeyEd25519> {
+    let index: u8 = ctx.parameter_cursor().get()?;
+    Ok(public_keys()[usize::from(index)])
+}
+
+/// View function that returns the balance of an validator. This is to follow ERC20 interface.
+#[cfg(any(feature = "production", feature = "development", feature = "sandbox"))]
+#[receive(
+    contract = "staking_bank",
+    name = "balanceOf",
+    parameter = "PublicKeyEd25519",
+    return_value = "u64"
+)]
+fn balance_of<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<u64> {
```

Comment:
Maybe it'd make sense to define 

```
type Amount = u64;
```
and use that everywhere instead of u64.

Reply:
Or perhaps a type alias which doesn't conflict with the CCD Amount ;)